lib/pbench/test/unit/server/test_schema.py

```diff
@@ -1,19 +1,67 @@
+from http import HTTPStatus
 import dateutil
 import pytest
 from typing import Callable
 
 from pbench.server.api.auth import Auth, UnknownUser
 from pbench.server.database.models.users import User
 from pbench.server.api.resources import (
+    API_OPERATION,
     ConversionError,
     InvalidRequestPayload,
     MissingParameters,
     Parameter,
     ParamType,
+    PostprocessError,
     Schema,
+    SchemaError,
+    UnauthorizedAccess,
+    UnsupportedAccessMode,
+    UnverifiedUser,
 )
 
 
+class TestExceptions:
+    """
+    Test exception stringification
+    """
+
+    def test_exceptions(self, create_user):
+        e = UnauthorizedAccess(create_user, API_OPERATION.READ, "you", "public")
+        assert (
+            str(e)
+            == "User test is not authorized to READ a resource owned by you with public access"
+        )
```

Comment:
Just a PSA:  I'm very sad to announce that, while this idiom should work with exception objects that you've created, it will _not_ work with exception objects which you have **caught** in a unit test, once we upgrade PyTest beyond V4.6.3.  (Peter gleefully pointed out that they've just released V6.2.5....)  This is because PyTest has been changed to provide the test code with an "exception info object" instead of an "exception object", and the info object has a slightly different stringification.

So, we need to migrate our practice to the following idiom:
```
assert "User test is not authorized to READ a resource owned by you with public access" in str(e)
```
so that the added "decoration" doesn't cause the assertion to fail when it shouldn't.

Reply:
Yes, I'm aware of that, and I've already ran into this. Not relevant here where I'm creating it directly.

However, I prefer to get at the real exception through the info object. I think it's `e.value`, but don't quote me on that.